### PR TITLE
Adds a protected flag to Roles.

### PR DESCRIPTION
**Pull request by zhalloran**
This pull request makes the following changes:

- Adds immutable "protected" field to the Role entity
- Modifies Role Authorizer to remove "delete" from allowed_privileges for protected roles
- Adds behat tests for protected roles

Test these changes by:
- Making authenticated DELETE requests against protected & unprotected roles

Fixes ushahidi/platform#1017, ushahidi/platform#1018, ushahidi/platform#1206 .

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform/1272)

### CODE REVIEW

**Comment by zhalloran** [comment]
@jasonmule When you get a chance, can you please do a code review of this pull request? Thanks!

File: migrations/20160623184725_set_protected_roles.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    /**
+     * Migrate Up.
+     */
+    public function up()
+    {
+        $this->execute("UPDATE roles
+        	SET protected = 1
+            WHERE
+                name = 'user' OR name = 'admin'
+            ");
```
Comment (rjmackay) [inline migrations/20160623184725_set_protected_roles.php]: Could this migration reinstate the user and admin roles if they don't exist too?

File: src/Core/Entity/Role.php [hunk 2]
```diff
@@ -48,6 +50,6 @@ public function getId()
 	// StatefulData
 	protected function getImmutable()
 	{
-		return ['name'];
+		return ['name','protected'];
```
Comment (rjmackay) [inline src/Core/Entity/Role.php]: 👍

File: application/tests/features/api.roles.feature [hunk 3]
```diff
@@ -94,3 +94,52 @@ Feature: Testing the Roles API
         And the "results.0.name" property equals "manager"
         Then the guzzle status code should be 200
 
+     Scenario: Get protected status of protected role
+        Given that I want to find a "Role"
+        And that the request "Authorization" header is "Bearer testadminuser"
+        And that its "id" is "1"
+        When I request "/roles"
+        Then the response is JSON
+		And the response has a "protected" property
```
Comment (rjmackay) [inline application/tests/features/api.roles.feature]: Your indenting seems to be off somehow? mixed tabs + spaces maybe.

**Comment by rjmackay** [comment]
This looks really good. Happy to land.

File: migrations/20160623184725_set_protected_roles.php (hunk 1 above)
Comment (zhalloran) [inline migrations/20160623184725_set_protected_roles.php]: Makes sense. I'll get that in today.

File: application/tests/features/api.roles.feature (hunk 3 above)
Comment (zhalloran) [inline application/tests/features/api.roles.feature]: Possible. I use Coda cuz I like some of the tooling for code syntax, change management, search and ssh, but the auto indentation "feature" on ⏎ can be a bit buggy.

**Pull request merged**